### ISSUE TITLE
Similar project

### ISSUE THREAD

**Issue by maandree** [colorjs/color-space#32]
Hi! I'm working on a similar C library.
https://github.com/maandree/libcolour

**Comment by dfcreative** [comment]
Awesome!
Feel free to take any code you may find useful.
Also if you have rare color-spaces - youre welcome to contribute!

**Comment by maandree** [comment]
I don't think I have any colour space you don't, except a lot of RGB colour spaces, but if I get any you don't have, I will let you know.

I do however have very exact numbers and not the rounded numbers you find everywhere.

Here are some things I have that might be interesting:

Standard illuminants with values calculated from the wavelength tables:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/libcolour.h#L65-L106

sRGB transfer function encoder/decoder with very exact values:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/common.h#L41-L65

Not sure if this will make any difference in JavaScript, but I have a bit more precise matrix for conversion from sRGB to CIE XYZ:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/matrices.py#L114-L122

A bunch of RGB colour spaces:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/libcolour-template.c#L362-L632
with their transfer functions:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/libcolour-template.c#L255-L332

And support for custom RGB colour spaces:
https://github.com/maandree/libcolour/blob/d9fd5eb7fbbcbf47c7398c591d7eda0ef2c5cce7/libcolour-template.c#L129-L227

**Comment by dfcreative** [comment]
Ok, that is a possible enhancement

**Comment by demurgos** [comment]
I agree that it would be great if you could clarify the situation around RGB color spaces. I'd say that most people refer to "sRGB" when they say "RGB" but there are many other variants such as linear RGB, CIE RGB or all the variants in maandree's project.

**Comment by maandree** [comment]
It is worth noting that is not uncommon that the term RGB is used to refer to CIE RGB without its transfer function.

**Comment by dy** [comment]
v3 did a full constants-precision pass: Bradford D50↔D65 consolidated at full precision (was 3 truncated copies), Lab ε=216/24389 & κ=24389/27, exact ACEScg/Rec.2020 matrices — all pinned by a differential suite vs colorjs.io. Will close when v3 (#67) merges.
